### PR TITLE
Filter media contribution list with perimeter

### PR DESCRIPTION
**Pull request by ngilain**
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/378
https://github.com/open-orchestra/open-orchestra-media-bundle/pull/258

### CODE REVIEW

File: MediaAdminBundle/Controller/Api/MediaController.php [hunk 1]
```diff
@@ -31,16 +32,22 @@ class MediaController extends BaseController
...
+     * @Config\Route("/list/with-perimeter/{siteId}", defaults={"withPerimeter": true}, name="open_orchestra_api_media_list_with_perimeter")
+     * @Config\Route("/list/without-perimeter/{siteId}", defaults={"withPerimeter": false}, name="open_orchestra_api_media_list_without_perimeter")
      * @Config\Method({"GET"})
      * @Config\Security("is_granted('IS_AUTHENTICATED_FULLY')")
      * @Api\Groups({MediaAdminGroupContext::MEDIA_ALTERNATIVES})
      *
      * @return FacadeInterface
      */
-    public function listAction(Request $request, $siteId)
+    public function listAction(Request $request, $siteId, $withPerimeter)
```
Comment (alavieille) [inline MediaAdminBundle/Controller/Api/MediaController.php]: phpdoc

File: MediaAdminBundle/Controller/Api/MediaController.php [hunk 2]
```diff
@@ -61,6 +68,30 @@ public function listAction(Request $request, $siteId)
...
+     * @param string $siteId
+     * @return null|array
+     */
+    protected function getConnectedUserMediaPerimeter($siteId)
+    {
+        $userGroups = $this->get('security.token_storage')->getToken()->getUser()->getGroups();
+        $folderIds = array();
+
+        foreach ($userGroups as $group) {
+            if ($group->hasRole('EDITORIAL_MEDIA_CONTRIBUTOR') && $group->getSite()->getSiteId() == $siteId) {
```
Comment (alavieille) [inline MediaAdminBundle/Controller/Api/MediaController.php]: use a constant

File: MediaAdminBundle/Controller/Api/MediaController.php [hunk 3]
```diff
@@ -61,6 +68,30 @@ public function listAction(Request $request, $siteId)
...
+                foreach ($group->getPerimeter(MediaFolderInterface::ENTITY_TYPE)->getItems() as $path) {
+                    foreach ($this->get('open_orchestra_media.repository.media_folder')->findSubTreeByPath($path) as $folder) {
```
Comment (alavieille) [inline MediaAdminBundle/Controller/Api/MediaController.php]: i think is better for the performance if the request is not in the foreach

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Collection/Media/Medias.js [hunk 4]
```diff
@@ -40,9 +40,16 @@ class Medias extends DataTableCollection
...
+        if ('read' == method && true == urlParameter.withoutPerimeter) {
+            method = 'read-without-perimeter';
+        }
+        delete urlParameter.withoutPerimeter;
+
         switch (method) {
             case "read":
-                return Routing.generate('open_orchestra_api_media_list', urlParameter);
+                return Routing.generate('open_orchestra_api_media_list_with_perimeter', urlParameter);
+            case "read-without-perimeter":
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Collection/Media/Medias.js]: i think is better to use the same mechanism than other model . and use the api context

**Comment by ngilain** [comment]
@alavieille : update done

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Collection/Media/Medias.js [hunk 5]
```diff
@@ -40,9 +40,15 @@ class Medias extends DataTableCollection
      */
     _getSyncUrl(method, options) {
         let urlParameter = options.urlParameter || {};
+        let apiContext = options.apiContext || {};
+
         switch (method) {
             case "read":
-                return Routing.generate('open_orchestra_api_media_list', urlParameter);
+                let route = 'open_orchestra_api_media_list_with_perimeter';
+                if (apiContext.withoutPerimeter) {
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Collection/Media/Medias.js]: why not use the string like other apiContext. If we want to add an other context, Without refactoring the view

**Comment by ngilain** [comment]
@alavieille : update done

**Comment by ngilain** [comment]
@alavieille : update done

**Review by alavieille — APPROVED** [review]

**Pull request merged**